apps/challenges/views.py

```diff
@@ -340,3 +343,56 @@ def challenge_phase_split_list(request, challenge_pk):
     serializer = ChallengePhaseSplitSerializer(result_page, many=True)
     response_data = serializer.data
     return paginator.get_paginated_response(response_data)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_all_submissions_of_challenge(request, challenge_pk, challenge_phase_pk):
+    """
+    Returns all the submissions for a particular challenge
+    """
+    # check if challenge exists or not.
+    challenge_model = get_model_object(Challenge)
+    challenge = challenge_model(challenge_pk)
+
+    # check if challenge phase exists or not.
```

Comment:
Instead of `check if challenge phase exists or not` what if we write the comment as `passed challenge as argument to make sure that this challenge phase belongs to the concerned challenge only`. This comment can be re-phrased in more better words. My idea here is that the comment should be less about what the code is, it should be more about why we need this or how are we using it further. Writing comments explaining what is not bad, but I feel that the code should be self explanatory.